bundles/org.openhab.binding.mideaac/src/main/java/org/openhab/binding/mideaac/internal/handler/MideaACHandler.java

```diff
@@ -0,0 +1,398 @@
+     * is valid. Because V2 devices do not require a cloud provider (or token/key)
+     * The first check is for the IP, port and deviceID. The second part
+     * checks the security configuration if required (V3 device).
+     */
+    @Override
+    public void initialize() {
+        if (isPollRunning) {
+            stopScheduler();
+        }
+        config = getConfigAs(MideaACConfiguration.class);
+
+        if (!config.isValid()) {
+            logger.warn("Configuration invalid for {}", thing.getUID());
+            if (config.isDiscoveryNeeded()) {
+                logger.warn("Discovery needed, discovering....{}", thing.getUID());
+                updateStatus(ThingStatus.UNKNOWN, ThingStatusDetail.CONFIGURATION_PENDING,
+                        "Configuration missing, discovery needed. Discovering...");
+                MideaACDiscoveryService discoveryService = new MideaACDiscoveryService();
+
+                try {
+                    discoveryService.discoverThing(config.ipAddress, this);
+                    return;
+                } catch (Exception e) {
+                    logger.error("Discovery failure for {}: {}", thing.getUID(), e.getMessage());
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                            "Discovery failure. Check configuration.");
+                    return;
+                }
+            } else {
+                logger.debug("MideaACHandler config of {} is invalid. Check configuration", thing.getUID());
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "Invalid MideaAC config. Check configuration.");
+                return;
+            }
+        } else {
+            logger.debug("Non-security Configuration valid for {}", thing.getUID());
+        }
+
+        if (config.version == 3 && !config.isV3ConfigValid()) {
+            if (config.isTokenKeyObtainable()) {
+                logger.info("Retrieving Token and/or Key from cloud");
+                CloudProviderDTO cloudProvider = CloudProviderDTO.getCloudProvider(config.cloud);
+                getTokenKeyCloud(cloudProvider);
+                return;
+            } else {
+                logger.warn("Configuration invalid for {} and no account info to retrieve from cloud", thing.getUID());
+                return;
+            }
+        } else {
+            logger.debug("Security Configuration (V3 Device) valid for {}", thing.getUID());
+        }
+
+        updateStatus(ThingStatus.UNKNOWN);
+
+        connectionManager = new org.openhab.binding.mideaac.internal.connection.ConnectionManager(config.ipAddress,
+                config.ipPort, config.timeout, config.key, config.token, config.cloud, config.email, config.password,
+                config.deviceId, config.version, config.promptTone);
+
+        startScheduler(2, config.pollingTime, TimeUnit.SECONDS);
```

Comment:
Reduce complexity and remove the startScheduler method too: 
```suggestion
        scheduledTask = scheduler.scheduleWithFixedDelay(this::pollJob, 2, config.pollingTime, TimeUnit.SECONDS);
```

Reply:
As noted above it has been a while.  However, I hate to mess with this section as I had issues with the original split of the connection manager with multiple schedulers getting started (one reason to kill the thread on line 220).  I need the logic in the startScheduler method.